cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1710,10 +1710,10 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
   size_type const total_frags = [&]() {
     if (frags_per_column.size() > 0) {
       std::exclusive_scan(frags_per_column.data(),
-                          frags_per_column.data() + num_columns + 1,
+                          frags_per_column.data() + num_columns,
```

Comment:
Let's use `frags_per_column.begin(), frags_per_column.end()`.

Reply:
Also please avoid `back_inserter` if possible. So this should be better:
```
std::vector<size_type> frag_offsets(num_columns, 0);
...
std::exclusive_scan(frags_per_column.begin(), frags_per_column.end(), frag_offsets.begin(), 0);
```

Reply:
much cleaner. thanks!

Reply:
AFAIK, the canonical way is to reserve the final size and use `back_inserter`, so we never reallocate and never default-initialize elements. It is unfortunatelly also the most verbose option :man_shrugging: 
Not a request to apply this here, just an FYI (or maybe a FMI, if @ttnghia knows of a `back_inserter` drawback).

Reply:
I like that. I wouldn't mind changing to that just so it's in my muscle memory :)